src/main/java/seedu/address/logic/parser/SearchCommandParser.java

```diff
@@ -0,0 +1,139 @@
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+        }
+
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+
+        String condition = argMultimap.getPreamble().toLowerCase();
+
+        switch (condition) {
+        case SearchCommand.AND_CONDITION:
+            return parseSearchWithAndCondition(argMultimap);
+        case SearchCommand.OR_CONDITION:
+            return parseSearchWithOrCondition(argMultimap);
+        case SearchCommand.EMPTY_CONDITION:
+            return parseSearchWithEmptyCondition(argMultimap);
+        default:
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+        }
+    }
+
+    private SearchCommand parseSearchWithAndCondition(ArgumentMultimap argMultimap) throws ParseException {
+        Pair<List<String>, List<List<String>>> keywordsAndPrefixes = extractPrefixesAndKeywords(argMultimap);
+        List<String> prefixes = keywordsAndPrefixes.getKey();
+        List<List<String>> keywords = keywordsAndPrefixes.getValue();
+        return new SearchCommand(new ContactContainsAllKeywordsPredicate(prefixes, keywords));
+    }
+
+    private SearchCommand parseSearchWithOrCondition(ArgumentMultimap argMultimap) throws ParseException {
+        Pair<List<String>, List<List<String>>> keywordsAndPrefixes = extractPrefixesAndKeywords(argMultimap);
+        List<String> prefixes = keywordsAndPrefixes.getKey();
+        List<List<String>> keywords = keywordsAndPrefixes.getValue();
+        return new SearchCommand(new ContactContainsAnyKeywordsPredicate(prefixes, keywords));
+    }
+
+    private SearchCommand parseSearchWithEmptyCondition(ArgumentMultimap argMultimap) {
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_NAME.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_NAME)));
+        } else if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_PHONE.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_PHONE)));
+        } else if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_EMAIL.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_EMAIL)));
+        } else if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_ADDRESS.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_ADDRESS)));
+        } else {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_TAG.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_TAG)));
+        }
+    }
```

Comment:
Yes what you state is true. I also realized the same thing regarding `EMPTY_CONDITION` can essentially be viewed as `AND_CONDITION` but not sure then how to go about that since it is planned in the user guide already. But I think your solution is great, to just treat `EMPTY_CONDITION` as `AND_CONDITION`. Thanks!